### PR TITLE
[Stacked 7] Finally upload the data to S3

### PR DESCRIPTION
**Pull request by ildyria**

### CODE REVIEW

**Comment by ildyria** [comment]
@Kovah could you check this PR if this is working for s3 ?  Pretty please.

You will need to set those in your `.env` :
```env
PHOTO_PIPES=true
S3_ENABLED=true
AWS_ACCESS_KEY_ID=
AWS_SECRET_ACCESS_KEY=
AWS_DEFAULT_REGION=
AWS_BUCKET=
AWS_URL=
AWS_ENDPOINT=
AWS_IMAGE_VISIBILITY=
AWS_USE_PATH_STYLE_ENDPOINT=
```

File: app/Actions/Photo/Pipes/Shared/UploadSizeVariantsToS3.php [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+	public function handle(PhotoDTO $state, \Closure $next): PhotoDTO
+	{
+		if (Features::active('use-s3')) {
+			$sync = Configs::getValueAsBool('use_job_queues');
+
+			$jobs = $state->getPhoto()->size_variants->toCollection()
+				->filter(fn ($v) => $v !== null)
+				->map(fn (SizeVariant $variant) => new UploadSizeVariantToS3Job($variant));
+
+			$jobs->each(fn ($job) => $sync ? dispatch($job) : dispatch_sync($job));
```
Comment (d7415) [inline app/Actions/Photo/Pipes/Shared/UploadSizeVariantsToS3.php]: This looks like the logic is the right way round, but the naming feels confusing - It looks like it uses asynchronous if true and synchronous if false?

**Review by d7415 — APPROVED** [review]

File: app/Actions/Photo/Pipes/Shared/UploadSizeVariantsToS3.php (hunk 1 above)
Comment (ildyria) [reply app/Actions/Photo/Pipes/Shared/UploadSizeVariantsToS3.php]: good point.

Comment (ildyria) [reply app/Actions/Photo/Pipes/Shared/UploadSizeVariantsToS3.php]: Fixed [7d2f7b3](https://github.com/LycheeOrg/Lychee/pull/2385/commits/7d2f7b356ca1359a6cd219e709eba574fa6568d6)

**Pull request merged**